tamr_unify_client/models/dataset/resource.py

```diff
@@ -94,6 +94,102 @@ def status(self) -> DatasetStatus:
             self.client, status_json, api_path=self.api_path + "/status"
         )
 
+    @property
+    def __geo_interface__(self):
+        """Retrieve a representation of this dataset that conforms to the Python Geo Interface.
+
+        Note that this materializes all features; for a streaming interface to features,
+        see :method:`~tamr_unify_client.models.dataset.Dataset.__geo_features__()`
+
+        See https://gist.github.com/sgillies/2217756
+
+        :return: dict[str, object]
+        """
+        return {
+            "type": "FeatureCollection",
+            "features": [feature for feature in self.iterfeatures()],
+        }
+
+    def iterfeatures(self):
+        """Returns an iterator that yields feature dictionaries that comply with __geo_interface__
+
+        See https://gist.github.com/sgillies/2217756
+
+        :return: stream of features
+        :rtype: Python generator yielding :py:class:`dict[str, object]`
+        """
+        key_attrs = self.key_attribute_names
+        if len(key_attrs) == 1:
+
+            def key_value(rec):
+                return rec[key_attrs[0]]
+
+        else:
+
+            def key_value(rec):
+                return [rec[attr] for attr in key_attrs]
+
+        # Duck-typing: find all the attributes that look like geometry
```

Comment:
Is that a real terminology, "Duck-typing". I get it, its funny. Looks like a duck, quacks like a duck. Just noting I had never heard this before :)

Reply:
https://en.wikipedia.org/wiki/Duck_typing

Reply:
FWIW, the Wikipedia article under-represents duck typing. For example, in this case we are pretty much explicitly stating that any attribute that _looks_ like it has a geometry type, _does_ have a geometry type. This is explicitly in place of having semantic types.